specs-go/v1/annotations.go

```diff
@@ -53,4 +53,10 @@ const (
 
 	// AnnotationDescription is the annotation key for the human-readable description of the software packaged in the image.
 	AnnotationDescription = "org.opencontainers.image.description"
+
+	// AnnotationBaseImageDigest is the annotation key for the digest of the image's base image.
+	AnnotationBaseImageDigest = "org.opencontainers.image.base.digest"
+
+	// AnnotationBaseImageRefName is the annotation key for the image reference of the image's base image.
+	AnnotationBaseImageRefName = "org.opencontainers.image.base.ref.name"
```

Comment:
This can drop the "ref". The "ref" portion was just extra hierarchy for containing the reference for the image itself. Since we are containing this under "base", that is sufficient:

```suggestion
	AnnotationBaseImageRefName = "org.opencontainers.image.base.name"
```

Reply:
.. e.g. docker pull base where base is specified by 
`[registry_host_name|IP address][:port][/v2][/org_path]<image_name>[:tag|@DIGEST]`
?

Reply:
> This can drop the "ref". The "ref" portion was just extra hierarchy for containing the reference for the image itself. Since we are containing this under "base", that is sufficient:

I'm fine with that if that's the consensus among image-spec maintainers/approvers.

> .. e.g. docker pull base where base is specified by
> `[registry_host_name|IP address][:port][/v2][/org_path]<image_name>[:tag|@DIGEST]`

The `image.base.[ref.]name` SHOULD be the reference _by tag_, not by digest. The `ref.name` should also be the image reference, not the HTTP(S) URL where the manifest can be found, so no `/v2/` path segment. This should already be specified in annotations.md, if there's a clearer way to describe that please feel free to suggest edits.

Reply:
ok... during the weekly call someone said it's the or like the docker pull format... so I put that ^ as a question during the call.. I see the commit diff adding the link to distribution/distribution.. for the formal format .. and note the differences between this and **org.opencontainers.image.ref.name** 's format... just trying to understand what's up here thus the question...

Reply:
Yeah, for the base image annotation it's intended to point to an image in a registry. For `image.ref.name` it _seems_ to be intended for use with image layouts, so that images in a layout structure can be location-addressable. I suppose a base image ref could also point to another image in a layout, and as I'm reading the grammar. I don't feel comfortable specifying that behavior though since I don't have much hands-on experience with layouts, and don't have a use case for it.